server/src/test/java/org/candlepin/resource/ConsumerResourceIntegrationTest.java

```diff
@@ -686,4 +694,179 @@ public void testContentAccessExpireRegen() {
+            .getRHCloudProfileModified();
+
+        Product prod = TestUtil.createProduct("Product One");
+        ConsumerInstalledProductDTO updatedInstalledProduct =
+            new ConsumerInstalledProductDTO(prod.getId(), prod.getName());
+        consumer.addInstalledProduct(updatedInstalledProduct);
+        consumerResource.updateConsumer(consumer.getUuid(), consumer, principal);
+
+        Date afterUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        assertNotNull(afterUpdateTimestamp);
+        assertNotEquals(beforeUpdateTimestamp, afterUpdateTimestamp);
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnHypervisorUpdate() {
+        ConsumerDTO consumer = createConsumerDTO("random-consumer", null, null, standardSystemTypeDTO);
+        consumer = consumerResource.create(consumer, principal, null, null, null, true);
+
+        Date beforeUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        HypervisorIdDTO updatedHypervisorId = new HypervisorIdDTO();
+        consumer.setHypervisorId(updatedHypervisorId);
+        consumerResource.updateConsumer(consumer.getUuid(), consumer, principal);
+
+        Date afterUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        assertNotNull(afterUpdateTimestamp);
+        assertNotEquals(beforeUpdateTimestamp, afterUpdateTimestamp);
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnSpecificConsumerFactsUpdate() {
+        ConsumerDTO consumer = createConsumerDTO("random-consumer", null, null, standardSystemTypeDTO);
+        consumer = consumerResource.create(consumer, principal, null, null, null, true);
+        Date modifiedDateOnCreate = consumerCurator.get(consumer.getId()).getRHCloudProfileModified();
+
+        ConsumerDTO updatedConsumerDTO = new ConsumerDTO();
+        updatedConsumerDTO.setFact("FACT", "FACT_VALUE");
+        consumerResource.updateConsumer(consumer.getUuid(), updatedConsumerDTO, principal);
+        Date modifiedTSOnUnnecessaryFactUpdate = consumerCurator.get(consumer.getId())
+            .getRHCloudProfileModified();
+
+        assertEquals(modifiedDateOnCreate, modifiedTSOnUnnecessaryFactUpdate);
+
+        updatedConsumerDTO = new ConsumerDTO();
+        updatedConsumerDTO.setFact(CloudProfileFacts.CPU_CORES_PERSOCKET.getFact(), "1");
+        consumerResource.updateConsumer(consumer.getUuid(), updatedConsumerDTO, principal);
+        Date modifiedTSOnNecessaryFactUpdate = consumerCurator.get(consumer.getId())
+            .getRHCloudProfileModified();
+
+        assertNotNull(consumerCurator.get(consumer.getId()).getRHCloudProfileModified());
+        assertNotEquals(modifiedTSOnUnnecessaryFactUpdate, modifiedTSOnNecessaryFactUpdate);
+    }
+
+    @Test
+    public void testCloudProfileFactDidNotChange() {
```

Comment:
This, and the 4 following tests which don't call into ConsumerResource, should probably moved into the `ConsumerTest.java` file, since that seems more relevant there

Reply:
Agreed, Moved these tests to `ConsmerTest.java`.